### PR TITLE
feat(eventing): Replica events

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Added replica creation and deletion events.
And Replica event specific data(pool name, pool uuid, replica name).

### CODE REVIEW

File: io-engine/src/lvs/lvs_lvol.rs [hunk 1]
```diff
@@ -1007,7 +1009,8 @@ impl LvsLvol for Lvol {
     async fn destroy_replica(mut self) -> Result<String, Error> {
         let snapshot_lvol = self.is_snapshot_clone();
         let name = self.name();
-        self.destroy().await?;
+        self.clone().destroy().await?;
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_lvol.rs]: hmm not sure if it's safe to clone here as we're deleting it, we might then later access memory which is now freed?
Might be better to put event on var first, and then after delete here then just call generate?

Comment (datacore-vvarakantham) [reply io-engine/src/lvs/lvs_lvol.rs]: Followed this now.

Also changed replica creation and deletion events to when "lvol" is created and deleted.

**Review by tiagolobocastro — APPROVED** [review]

**Review by chriswldenyer — APPROVED** [review]
LGTM

**Comment by datacore-vvarakantham** [comment]
bors merge

**Pull request merged**